lib/timber/events/channel_receive_event.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Timber.Events.ChannelReceiveEvent do
+  @moduledoc """
+  The `ChannelReceiveEvent` represents the reception of an event for a given topic on a channel.
+
+  The defined structure of this data can be found in the log event JSON schema:
+  https://github.com/timberio/log-event-json-schema
+  """
+
+  @type t :: %__MODULE__{
+    channel: String.t,
+    topic: String.t,
+    event: String.t,
+    metadata_json: String.t | nil,
+  }
+
+  @enforce_keys [:channel, :topic, :event]
+  defstruct [
+    :channel,
+    :topic,
+    :event,
+    :metadata_json
+  ]
+
+  @metadata_json_byte_limit 8_192
+
+  @doc """
+  Builds a new struct taking care to:
+
+  * Converts `:params` to `:params_json` that satifies the Timber API requirements
+  """
+  @spec new(Keyword.t) :: t
+  def new(opts) do
+    metadata_json =
+      case Keyword.get(opts, :metadata_json, nil) do
+        nil ->
+          nil
+
+        metadata_json ->
+          metadata_json
+          |> Timber.Utils.Logger.truncate_bytes(@metadata_json_byte_limit)
+          |> to_string()
+      end
+
+    new_opts = Keyword.put(opts, :metadata_json, metadata_json)
+
+    struct!(__MODULE__, new_opts)
+  end
+
+  @doc """
+  Message to be used when logging.
+  """
+  @spec message(t) :: IO.chardata
+  def message(%__MODULE__{channel: channel, topic: topic, event: event}) do
+    ["Incoming ", inspect(event), " on ", to_string(topic), " to ", to_string(channel)]
```

Comment:
1. "Incoming" is a present participle used as an adjective to describe "event". This breaks with our norm of using the simple past form of a transitive verb with the event description as the object (e.g., ["Processed"](https://github.com/timberio/timber-elixir/blob/master/lib/timber/events/sql_query_event.ex#L25) or ["Received"](https://github.com/timberio/timber-elixir/blob/master/lib/timber/events/http_request_event.ex#L98)). I recommend we change this to "Received" to keep with precedent.
2. I don't understand why `inspect/1` is being used on `event`. `event` should already be a string, as specified by the typespec. If there is a possibility it will be an atom, then using `to_string/1` should result in a string regardless of whether event is an atom or string. We should avoid any unnecessary use of `inspect/1`.